docs/data_usage_guides/metadata.md

```diff
@@ -0,0 +1,166 @@
+# Metadata
+
+## HDX Metadata
+
+All of the data in HDX HAPI comes from publicly-available datasets on HDX, and
+the HDX metadata can be used refer back to the original data to see what HAPI
+has simplified or omitted.
+The two primary HDX metadata tables are
+[`dataset`](dataset) and [`resource`](resource), and they
+contain a small subset of the metadata available through the CKAN API or
+HDX library.
+
+The HAPI data records contain enough information to access the full records
+from CKAN, see the parameters tables for more details.
+All subcategory tables link back to a resource record, and all resources link
+back to a dataset record.
+
+### Dataset <a id="dataset"></a>
+
+**Used in:** [`Resource`](resource), all sub-category tables
+
+This table contains the HDX-specific metadata associated with all datasets
+used to populate the HAPI sub-category tables. Every dataset has at least
+one child [resource](resource).
+
+### Resource <a id="resource"></a>
+
+**Used in:** All sub-category tables
+
+This table contains the HDX-specific metadata associated with all resources
+used to populate the HAPI sub-category tables. Every resource has one
+parent [dataset](dataset).
+
+## Geographical Metadata
+
+HAPI supports three hierarchical levels of geographical metadata:
+location (a country or country-like entity), admin 1, and admin 2.  An entry in
+the location table does not necessarily imply UN recognition of statehood.
+The subcategory data tables link to the lowest administrative level used by
+that data type; it will usually be admin2, but in some cases may be admin1 or
+location.
+
+The names and p-codes are read in from a [global p-code list](https://data.humdata.org/dataset/global-pcodes)
+taken from the common operational dataset (COD) gazetteers administrative
+boundaries.
+The p-code field is unique only in combination with reference_period_start,
+since p-codes may be reused in different versions of geographical metadata.
+
+When bringing in data from sub-categories, HDX HAPI uses p-code matching
+from [`hdx-python-country`](https://hdx-python-country.readthedocs.io/en/latest/).
+Thie can involve .... TODO
```

Comment:
HDX HAPI utilises p-code handling provided by the [`hdx-python-country`](https://hdx-python-country.readthedocs.io/en/latest/) library. When bringing in data from sub-categories, p-codes are used if present. Sometimes, these p-codes may not strictly follow the standard with variations such as shorter or longer country or admin unit codes. The hdx-python-country library has a p-code length matching algorithm to ensure correct admin units are determined. Where p-codes are unavailable in the data, it has the capability to match admin unit names to p-codes aided by both automatic phonetic matching and manual overrides.

Reply:
amazing thank you

Reply:
@IanHopkinson there's a storymap [here] (https://storymaps.arcgis.com/stories/dcf6135fc0e943a9b77823bb069e2578) that describes the CODs and p-codes!